acm/src/main/java/org/literacybridge/acm/gui/ResourceView/audioItems/AudioItemTableModel.java

```diff
@@ -2,57 +2,134 @@
+            File file = ACMConfiguration.getInstance().getCurrentDB().getRepository().getAudioFile(audioItem, AudioFormat.A18);
+            if (file != null) {
+                Date date = new Date(file.lastModified());
+                value = DATE_FORMAT.format(date);
+            }
 
-    // positions of the table columns
-    public static final int NUM_COLUMNS 	   = 8; // keep in sync
-    public static final int INFO_ICON 		   = 0;
-    public static final int TITLE 			   = 1;
-    public static final int DURATION 		   = 2;
-    public static final int CATEGORIES 		   = 3;
-    public static final int SOURCE			   = 4;
-    public static final int LANGUAGES          = 5;
-    public static final int DATE_FILE_MODIFIED = 6;
-    public static final int PLAYLIST_ORDER	   = 7;
-    private static String[] columns = null;
+            return new AudioItemNode<String>(audioItem, value);
+        }
+    });
+    public static final ColumnInfo<Integer> PLAYLIST_ORDER_COLUMN = ColumnInfo.newColumnInfo(LabelProvider.AUDIO_ITEM_TABLE_COLUMN_PLAYLIST_ORDER, ColumnInfo.WIDTH_NOT_SET, 60, new ValueProvider<Integer>(false) {
+        @Override
+        protected AudioItemNode<Integer> getValue(AudioItem audioItem) {
+            Playlist tag = Application.getFilterState().getSelectedPlaylist();
+            int position = 0;
+            if (tag != null) {
+                position = tag.getAudioItemPosition(audioItem.getUuid()) + 1;
+            }
+            return new AudioItemNode<Integer>(audioItem, position);
+        }
+    }).setComparator(new Comparator<AudioItemNode<Integer>>() {
+        @Override
+        public int compare(AudioItemNode<Integer> o1,
+                AudioItemNode<Integer> o2) {
+            return Integer.compare(o1.getValue(), o2.getValue());
+        }
+    });
 
-    protected SearchResult result = null;
     private final MetadataStore store;
 
-    public static void initializeTableColumns(String[] initalColumnNames) {
-        columns = initalColumnNames;
-    }
+    private final Map<String, Integer> uuidToRowIndexMap;
+    private final List<AudioItemNodeRow> rowIndexToUuidMap;
+
+    private final ColumnInfo<?>[] columns;
 
-    public AudioItemTableModel(SearchResult result) {
-        this.result = result;
+    public AudioItemTableModel() {
         this.store = ACMConfiguration.getInstance().getCurrentDB().getMetadataStore();
-        if (result != null) {
-            result.getAudioItems();
+        this.uuidToRowIndexMap = Maps.newHashMap();
+        this.rowIndexToUuidMap = Lists.newArrayList();
+
+        columns = initializeColumnInfoArray(
+                INFO_ICON_COLUMN,
+                TITLE_COLUMN,
+                DURATION_COLUMN,
+                CATEGORIES_COLUMN,
+                SOURCE_COLUMN,
+                LANGUAGES_COLUMN,
+                DATE_FILE_MODIFIED_COLUMN,
+                PLAYLIST_ORDER_COLUMN);
+
+        // iterate over all audioItems and add them to the uuid->rowIndex map
```

Comment:
Comment simply repeats the code.  

Comments should mostly answer "why is this code here?" or "why is it done THIS way?".  When we need a comment for "what does this code do?" we probably need to rewrite for clarity; sometimes that's not possible, and in that case a "what does this do" is necessary.